webui/react/src/components/Section.test.tsx

```diff
@@ -0,0 +1,71 @@
+import { render, screen } from '@testing-library/react';
+import React from 'react';
+
+import Section from './Section';
+
+const setup = ({
+  title = '',
+  loading = false,
+  hideTitle = false,
+  options = <div />,
+  filters = <div />,
+  maxHeight = false,
+  bodyBorder = false,
+  bodyNoPadding = false,
+  divider = false,
+}) => {
+  const handleOnChange = jest.fn();
+  const view = render(
+    <Section
+      bodyBorder={bodyBorder}
+      bodyNoPadding={bodyNoPadding}
+      divider={divider}
+      filters={filters}
+      hideTitle={hideTitle}
+      loading={loading}
+      maxHeight={maxHeight}
+      options={options}
+      title={title}
+    />,
+  );
+  return { handleOnChange, view };
+};
+
+describe('Section', () => {
+
+  it('Section with title', () => {
+    setup({ title: 'title of section' });
+    expect(screen.getByText('title of section')).toBeInTheDocument();
+  });
+
+  it('Section hide title', () => {
+    setup({ hideTitle: true, title: 'title of section' });
+    expect(screen.queryAllByText('title of section')).toHaveLength(0);
+  });
+
+  it('Section in loading state', () => {
+    setup({ loading: true, title: 'section-title' });
+    expect(screen.getByText('section-title')).toBeInTheDocument();
+  });
```

Comment:
Could this test instead have two expects: checking the Spinner component prop spinning=true, and check that section-filters are hidden?